oracledb-ballerina/types.bal

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/sql;
+
+
+# IntervalYearToMonth stores a period of time in years and months.
+#
+# + years - Number of years
+# + months - Number of months
+type IntervalYearToMonth record {|
+    int years;
+    int months;
+|};
+
+
+# IntervalDayToSecond stores a period of time in days, hours, minutes, and seconds.
+#
+# + days - Number of days
+# + hours - Number of hours
+# + minutes - Number of minutes
+# + seconds - Number of seconds
+type IntervalDayToSecond record {|
+    int days;
+    int hours;
+    int minutes;
+    decimal seconds;
+|};
+
+
+# ObjectType is an abstraction of the real-world entities, such as purchase orders, that application programs deal with.
+#
+# + typename - Name of the object type
+# + attributes - Attributes of the object
+type ObjectType record {|
+    string typename;
+    anydata[]? attributes;
+|};
+
+
+# Varray is an ordered set of data elements with a variable size. All elements of a given array are of the same data
+# type.
+#
+# + name - Name of the varray
+# + elements - Elements of the Varray
+type Varray record {|
+    string name;
+    byte[]|int[]|boolean[]|float[]|decimal[]|string[]? elements;
+|};
+
+
+# NestedTable models an unordered set of elements. The elements may be built-in types or user-defined types.
+#
+# + name - Name of the varray
+# + attributes - Attributes of the nested table
+type NestedTable record {|
+    string name;
+    anydata[] attributes;
+|};
```

Comment:
As discussed, shall we not support nested tables for the initial phase

Reply:
Removed with e4b5269039da7c79bb26819d3e7c23dae72290f4